src/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as core from "@actions/core";
+import { File as DiffFile } from "parse-diff";
+import minimatch from "minimatch";
+import { callModel, createReviewPrompt, parseAIResponse } from "./ai";
+import { ReviewComment } from "./types";
+import {
+  getFileContent,
+  getPRContext,
+  getRepoContext,
+  submitReviewComments,
+} from "./git";
+
+async function prepareReviewContext(
+  prContext: Awaited<ReturnType<typeof getPRContext>>
+) {
+  const { pr, files, parsedDiff, owner, repo } = prContext;
+
+  const repoContext = await getRepoContext(owner, repo);
+  const excludePatterns = core
+    .getInput("exclude")
```

Comment:
Consider validating the input for 'exclude' to ensure it is not empty or malformed before processing it.